ts/a11y/SpeechMenu.ts

```diff
@@ -0,0 +1,209 @@
+ */
+
+import { SpeechExplorer } from './explorer/KeyExplorer.js';
+import { ExplorerMathItem } from './explorer.js';
+import {MJContextMenu} from '../ui/menu/MJContextMenu.js';
+import {Submenu} from 'mj-context-menu/js/item_submenu.js';
+import Sre from './sre.js';
+
+/**
+ * Values for the ClearSpeak preference variables.
+ */
+let csPrefsSetting: {[pref: string]: string} = {};
+
+/**
+ * Generator of all variables for the Clearspeak Preference settings.
+ * @param {MJContextMenu} menu The current context menu.
+ * @param {string[]} prefs The preferences.
+ */
+function csPrefsVariables(menu: MJContextMenu, prefs: string[]) {
+  let srVariable = menu.pool.lookup('speechRules');
+  let previous = Sre.clearspeakPreferences.currentPreference();
+  csPrefsSetting = Sre.clearspeakPreferences.fromPreference(previous);
+  for (let pref of prefs) {
+    menu.factory.get('variable')(menu.factory, {
+      name: 'csprf_' + pref,
+      setter: (value: string) => {
+        csPrefsSetting[pref] = value;
+          srVariable.setValue(
+          'clearspeak-' +
+              Sre.clearspeakPreferences.toPreference(csPrefsSetting)
+        );
+      },
+      getter: () => { return csPrefsSetting[pref] || 'Auto'; }
+    }, menu.pool);
+  }
+}
+
+/**
+ * Generate the selection box for the Clearspeak Preferences.
+ * @param {MJContextMenu} menu The current context menu.
+ * @param {string} locale The current locale.
+ */
+function csSelectionBox(menu: MJContextMenu, locale: string) {
+  let prefs = Sre.clearspeakPreferences.getLocalePreferences();
+  let props = prefs[locale];
+  if (!props) {
+    let csEntry = menu.findID('Accessibility', 'Speech', 'Clearspeak');
+    if (csEntry) {
+      csEntry.disable();
+    }
+    return null;
+  }
+  csPrefsVariables(menu, Object.keys(props));
+  let items = [];
+  for (const prop of Object.getOwnPropertyNames(props)) {
+    items.push({
+      'title': prop,
+      'values': props[prop].map(x => x.replace(RegExp('^' + prop + '_'), '')),
+      'variable': 'csprf_' + prop
+    });
+  }
+  let sb = menu.factory.get('selectionBox')(menu.factory, {
+    'title': 'Clearspeak Preferences',
+    'signature': '',
+    'order': 'alphabetic',
+    'grid': 'square',
+    'selections': items
+  }, menu);
+  return {'type': 'command',
+          'id': 'ClearspeakPreferences',
+          'content': 'Select Preferences',
+          'action': () => sb.post(0, 0)};
+}
+
+/**
+ * Generates the menu items for the base preference menu.
+ *
+ * @param {string} previous The currently set preferences.
+ * @return The menu items.
+ */
+function basePreferences(previous: string) {
+  const items = [
+      {
+        type: 'radio',
+        content: 'No Preferences',
+        id: 'clearspeak-default',
+        variable: 'speechRules'
+      },
+      {
+        type: 'radio',
+        content: 'Current Preferences',
+        id: 'clearspeak-' + previous,
+        variable: 'speechRules'
+      },
+      { type: 'rule' },
```

Comment:
Again, we haven't uses extra spaces in this setting.  Perhaps spread out the object over three lines?

Reply:
Code moved over from SRE, so that is the automatic formatting by `prettier`.

Reply:
Also fixed the indentation.